code/controllers/subsystem/processing/objects.dm

```diff
@@ -22,29 +21,34 @@ var/datum/subsystem/objects/SSobj
 
 /datum/subsystem/objects/Initialize(timeofdayl)
 	fire_overlay.appearance_flags = RESET_COLOR
-	trigger_atom_spawners()
 	setupGenetics() //to set the mutations' place in structural enzymes, so monkey.initialize() knows where to put the monkey mutation.
-	for(var/thing in world)
-		var/atom/A = thing
-		A.Initialize(TRUE)
-		CHECK_TICK
-	initialized = TRUE
+	InitializeAtoms()
 	. = ..()
 
+/datum/subsystem/objects/proc/InitializeAtoms(list/objects = null)
+	initialized = 1
+
+	if(objects)
+		for(var/thing in objects)
+			var/atom/A = thing
+			A.Initialize(TRUE)
+			CHECK_TICK
+	else
+		for(var/thing in world)
```

Comment:
IIRC var/thing in world included datums

that being said, typeless for loops aren't faster when accessing world or contents and typing to a built in type because byond stores seperate lists and just changes what list it accesses rather then istype everything.

Changing this to `for(var/atom/thing in world)` would actually be faster because of the in world snowflake byond has.